### PR TITLE
ipatests: add test for AD users should not lose their AD domains when it is added to idview.

### PR DESCRIPTION
**Pull request by amore17**
When AD user is added with customized idview.
Then SSSD should not fail to retrieve  AD domain details.

Related : https://pagure.io/SSSD/sssd/issue/4173

### CODE REVIEW

**Comment by fcami** [comment]
hi @amore17 this fails with:
```
./ipatests/test_integration/test_sssd.py:397:81: E501 line too long (85 > 80 characters)
```
I think PR-CI was triggered on the wrong commit though. Could you:
- rebase and remove the temp commit from your branch
- add a temp commit at the top of your branch
- push -f

Thanks

File: ipatests/test_integration/test_sssd.py [hunk 1]
```diff
@@ -387,6 +387,42 @@ def test_trustdomain_disable_does_not_disable_root_domain(self,
...
+        1. Override AD user's UID, GID by adding it in ID view on IPA server.
+        2. Stop the SSSD, and clear SSSD cache and restart SSSD on a IPA client
+        3. getent with UID added from ID view should return AD domain after 6minutes.
+        """
+        client = self.clients[0]
+        user = self.users['ad']['name']
+        idview = 'testview'
+
+        def verify_retrieved_users_domain():
+            test_user = ['su', user, '-c', 'sleep 360; getent passwd 10001']
```
[the remark this answers is not in the document]
Comment (amore17) [reply ipatests/test_integration/test_sssd.py:404]: The duration is not linked to a timeout set in sssd.conf.

[the remark this answers is not in the document]
Comment (flo-renaud) [reply ipatests/test_integration/test_sssd.py:404]: I quickly checked with SSSD developers, and the issue happens after ~5 min which is sssd's memcache_timeout. Could you add a comment here, like for instance:
```
Wait for the record to expire in SSSD's cache (memcache_timeout default value is 300s).
```

[the remark this answers is not in the document]
Comment (amore17) [reply ipatests/test_integration/test_sssd.py:407]: Thanks for looking. Added comment.

**Comment by flo-renaud** [comment]
Hi @amore17 
the logs for [fedora 32 with sssd 2.2.3](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/c0eafeb2-b6c8-11ea-a56b-fa163e43b58c/report.html) show that the test properly detects the issue, and the logs for [fedora 32 with sssd 2.3.0](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/f7b37b60-b482-11ea-a359-fa163e43b58c/report.html) show that the issue is fixed with sssd 2.3.0
Please remove the temp commit and I will ACK when PR-CI completes.

**Comment by varunmylaraiah** [comment]
Look good to me.

**Comment by flo-renaud** [comment]
master:

* 0cb0056fd6fc3f8665b11a196d8c613926d40a95 ipatests: Test that trusted AD users should not lose their AD domains.
* d39786c0514b124f0e11a3fbcfef104d1eccea3b ipatests: xfail test with older versions of sssd

**Pull request closed without merging**